src/util/filesystem.cpp

```diff
@@ -1,6 +1,107 @@
 #include <util/filesystem.h>
 #include <boost/filesystem.hpp>
 
+using namespace file_operations;
+
+tmp_path::tmp_path(std::string path, bool keep)
+  : _path(std::move(path)), _keep(keep)
+{
+  assert(boost::filesystem::exists(_path));
+}
+
+tmp_path::tmp_path(tmp_path &&o) : tmp_path(std::move(o._path), o._keep)
+{
+  o._keep = true;
+}
+
+tmp_path::~tmp_path()
+{
+  if(_keep)
+    return;
+  [[maybe_unused]] uintmax_t removed = boost::filesystem::remove_all(_path);
```

Comment:
If I am not mistaken, this is a C++17 directive. For the project we still rely on C++14, you can make this target use C++17 inside cmake.

Reply:
I've seen [[gnu::unused]] elsewhere in the code and used that, ok?

Reply:
It should be okay.